src/core/plugin/luapi_application.h

```diff
@@ -2188,6 +2211,81 @@ static int applib_addImagesFromFilepath(lua_State* L) {
     return static_cast<int>(cntParams);
 }
 
+static int applib_getImages(lua_State* L) {
+    Plugin* plugin = Plugin::getPluginFromLua(L);
+    std::string type = luaL_checkstring(L, 1);
+    std::vector<Element*> elements = {};
+    Control* control = plugin->getControl();
+
+    // Discard any extra arguments passed in
+    lua_settop(L, 1);
+    luaL_checktype(L, 1, LUA_TSTRING);
```

Comment:
This is kinda duplicated with `std::string type = luaL_checkstring(L, 1);` in `2216`. `lua_L_checkstring` already errors if the argument is not a string. But this is something we're doing "wrong" at some (most other `get`... functions) other places as well (not sure if I copied this from somewhere or if I introduced this by myself). Gonna need to cleanup this in the next time.

As a matter of facts in my tests, neither `luaL_checkstring` nor `luaL_checktype(L, 1, LUA_TSTRING);` errored when passing a number like `1` (got just converted to `"1"` as far as I saw it, and then errored with `"Unknown argument"` in the if-then-else below), but when passing a boolean like `true` it did.
I think this might be a bug on the lua-side (do you have this as well?) and there's nothing we can immediately do about it.

Reply:
It's not a bug on the Lua side, it's a feature. :-)

```c++
int lua_isstring (lua_State *L, int index);

Returns 1 if the value at the given index is a string or a number (which is always convertible to a string), and 0 otherwise. 
```